src/archivematicaCommon/lib/elasticSearchFunctions.py

```diff
@@ -78,15 +78,74 @@
     'index': 'not_analyzed'
 }
 
+
 class ElasticsearchError(Exception):
+    """ Not operational errors. """
     pass
 
+
 class EmptySearchResultError(ElasticsearchError):
     pass
 
+
 class TooManyResultsError(ElasticsearchError):
     pass
 
+
+def setup(hosts):
+    """
+    Initialize Elasticsearch client and share it as the attribute _es_client in
+    the current module. An additional attribute _es_hosts is defined containing
+    the Elasticsearch hosts (expected types are: string, list or tuple).
+    """
+    module = sys.modules[__name__]
+    setattr(module, '_es_hosts', hosts)
+    setattr(module, '_es_client', Elasticsearch(**{'hosts': _es_hosts}))
+
+
+def setup_reading_from_client_conf(config=None):
+    """
+    Similar to setup() but added to guarantee backward compatibility with old
+    consumers. Load the configuration from clientConfig.conf. It also accepts
+    an instance of ConfigParser, which is convenient when the consumer already
+    has loaded the configuration for other reasons.
+    """
+    if not config:
+        config = ConfigParser.SafeConfigParser()
+        config.read('/etc/archivematica/MCPClient/clientConfig.conf')
+    try:
+        hosts = config.get('MCPClient', "elasticsearchServer")
+    except (ConfigParser.NoSectionError, ConfigParser.NoOptionError):
+        hosts = '127.0.0.1:9200'
+    setup(hosts)
+
+
+def get_host():
+    """
+    Return one of the Elasticsearch hosts configured in our client. In the
+    future this function could look it up in the Elasticsearch client instead
+    of using the module attribute _es_hosts, because in an Elasticsearch
+    cluster, nodes can be added or removed dynamically.
+    """
+    if not _es_hosts:
+        raise ImproperlyConfigured('The Elasticsearch client has not been set up yet.')
```

Comment:
Update this to mention the correct setup function? Eg `raise ImproperlyConfigured('The Elasticsearch client has not been set up yet. Please call setup() first')`